tools/IoTEdgeDevOps/DevOpsLib/BuildManagement.cs

```diff
@@ -21,25 +21,33 @@ public BuildManagement(DevOpsAccessSetting accessSetting)
             this.accessSetting = accessSetting;
         }
 
-        public async Task<VstsBuild> GetLatestBuildAsync(int buildDefinitionId, string branchName)
+        public async Task<IDictionary<int, VstsBuild>> GetLatestBuildsAsync(int[] buildDefinitionIds, string branchName)
         {
-            string requestPath = string.Format(LatestBuildPathSegmentFormat, this.accessSetting.Organization, this.accessSetting.Project, buildDefinitionId);
+            ValidationUtil.ThrowIfNulOrEmptySet(buildDefinitionIds, nameof(buildDefinitionIds));
+            ValidationUtil.ThrowIfNullOrWhiteSpace(branchName, nameof(branchName));
+
+            string requestPath = string.Format(LatestBuildPathSegmentFormat, this.accessSetting.Organization, this.accessSetting.Project);
             IFlurlRequest latestBuildRequest = DevOpsAccessSetting.BaseUrl
                 .AppendPathSegment(requestPath)
-                .SetQueryParam("api-version", "5.1-preview.1")
+                .SetQueryParam("definitions", string.Join(",", buildDefinitionIds))
+                .SetQueryParam("queryOrder", "finishTimeDescending")
+                .SetQueryParam("maxBuildsPerDefinition", "1")
+                .SetQueryParam("api-version", "5.1")
                 .SetQueryParam("branchName", branchName)
                 .WithBasicAuth(string.Empty, this.accessSetting.PersonalAccessToken);
 
             string resultJson = await latestBuildRequest.GetStringAsync().ConfigureAwait(false);
```

Comment:
would you please point me clearly which section talk about onErrorAsync usage on that link?

Reply:
Per in-person discussion, could you please add a TODO here so that we don't forget to link it up with once we decide on the Exception handling scheme.